src/lib.rs

```diff
@@ -152,25 +152,30 @@ use std::{
     convert::TryInto,
     fmt,
     os::unix::io::RawFd,
+    str::FromStr,
     str, thread,
     time::SystemTime,
     vec,
 };
 
 /// A basic error type from this library.
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
```

Comment:
I was thinking if we should use `#[non_exhaustive]` here.  I don't have enough experience to make a definitive choice. I will leave as is.

Reply:
good catch, error variants definitely should be `#[non_exhaustive]`